nwb_conversion_tools/utils/spike_interface.py

```diff
@@ -273,8 +273,8 @@ def add_electrodes(
     else:
         checked_recording = recording
     if nwbfile.electrodes is not None:
-        ids_absent = [id not in nwbfile.electrodes.id for id in checked_recording.get_channel_ids()]
-        if not all(ids_absent):
+        ids_already_in_electrode_table = [id in nwbfile.electrodes.id for id in checked_recording.get_channel_ids()]
+        if all(ids_already_in_electrode_table):
             warnings.warn("cannot create electrodes for this recording as ids already exist")
             return
```

Comment:
I was thinking about this.
@CodyCBakerPhD @Saksham20 

In the previous version in the diff we **only write the electrodes table if ALL the channels are new**. That is, we only write the electrodes table if NONE of the channels in the recording object were previously available in the electrodes table. In other words, if there is only one or two channels that are not in the electrodes table this not enough for us to add them to the table.

Am I misunderstanding? 
Is this the desired behavior? 
My naive understanding is that if there are channels whose properties were not written before maybe we should add them to the table? 

The version that I implemented in this change does that: **the only way you don't write the electrode table is if ALL the channels you are adding were previously available in the electrode table**. That is, there is no new channel in the recording object.

Reply:
> The version that I implemented in this change does that: the only way you don't write the electrode table is if ALL the channels you are adding were previously available in the electrode table. That is, there is no new channel in the recording object.

This should indeed be the desired behavior. I do agree the previous logic would have falsely terminated if _some_ channels were and _some_ channels weren't present in the existing table.

Reply:
Good catch!